packages/frontend/src/features/sqlRunner/components/visualizations/BarChartConfiguration.tsx

```diff
@@ -0,0 +1,82 @@
+import { type BarChartConfig } from '@lightdash/common';
+import { Stack, Title } from '@mantine/core';
+import debounce from 'lodash/debounce';
+import { type FC } from 'react';
+import { EditableText } from '../../../../components/VisualizationConfigs/common/EditableText';
+
+type BarChartConfigurationProps = {
+    value: BarChartConfig;
+    onChange: (config: BarChartConfig) => void;
+};
+
+const DEBOUNCE_TIME = 500;
+
+const BarChartConfiguration: FC<BarChartConfigurationProps> = ({
+    value,
+    onChange,
+}) => {
+    const onXAxisLabelChange = debounce((label: string) => {
+        onChange({
+            ...value,
```

Comment:
Doing all of this spreading before sending the values to the store seems to me like skirting some of the point of using RTK. You were concerned about having too many specific setters for individual properties, no? But it seems like this way we have moved that complexity out of the slice and now we have to do this error-prone spreads in these configuration files for every property. I don't think this is less code than the way we were doing it before where there were actions per-property, right? And I think it's probably slower and more error prone than passing one property to the store and letting relying on RTK/Immer to set the nested properties. One of the things RTK is good at/good for is surgically alerting object without having to spread/copy the whole thing.

Maybe I missed a reason to do this 😅? If we do want to only have one setter in the store that takes the whole config, maybe we should use Immer here?